advanced_functionality/multi_model_linear_learner_home_value/linear_learner_multi_model_endpoint_inf_pipeline.ipynb

```diff
@@ -0,0 +1,1302 @@
+{
```

Comment:
<p>Considering that <code>role_name</code> is fixed, does this cell work if the notebook is executed repeatedly?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/awslabs/amazon-sagemaker-examples/pull/1311/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='67'/>

Reply:
<p>Added date/time string to role name to make this unique.</p><p> </p><p>role_name='allow_invoke_ny_model_role'+strftime('%Y-%m-%d-%H-%M-%S', gmtime())</p>

Reply:
Is it the best option? With this change in place, the notebook will create a new role every time it is executed. An error can be handled gracefully instead.

```python
try:
    response = iam_client.create_role(
        Path=path,
        RoleName=role_name,
        AssumeRolePolicyDocument=json.dumps(trust_policy),
        Description=description,
        MaxSessionDuration=3600
    )
except iam_client.exceptions.EntityAlreadyExistsException:
    print("Role '{}' already exists.".format("role_name"))
```

Other parts of the notebook may need to be changed as well, e.g. to not attach a policy if role exists. There may be different ways to do that. One option is, move the functionality to functions.

```python
def create_new_role_and_attach_granular_policy(role_name):
    try:
        response = iam_client.create_role(
            Path=path,
            RoleName=role_name,
            AssumeRolePolicyDocument=json.dumps(trust_policy),
            Description=description,
            MaxSessionDuration=3600
        )
    except iam_client.exceptions.EntityAlreadyExistsException:
        print("Role '{}' already exists.".format("role_name"))
        return

    policy = create_granular_policy()  # needs to be created
    iam_client.attach_role_policy(PolicyArn=policy_arn, RoleName=role_name)

create_new_role_and_attach_granular_policy("allow_invoke_ny_model_policy")
```

See also what works best from clarity and readability perspectives. Feel free to keep as is if you prefer.

Reply:
Going to leave this in as is for better readability